sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,514 @@
+            eff_area = (var['geometric_area_ccd'] * u.cm ** 2) * reflectance * transmission_efficiency * var[
+                'ccd_contamination'][channel_index] * var['quantum_efficiency_ccd'][channel_index]
+            self.effective_area = eff_area
+
+            if print_comparison_ratio:
+                # used for testing - move to test_aia?
+                compare_eff_area = [0.312, 1.172, 2.881, 1.188, 1.206, 0.063, 0.045, 0.0192, 0.0389][
+                    self.channel_list.index(channel)]
+                print('eff_area: ', compare_eff_area, eff_area, 'ratio of comparison: ', eff_area / compare_eff_area)
+
+    def calculate_system_gain(self, channel, use_table2_gain=False, use_genx_values=False):
+        """
+
+        :return:
+        """
+
+        var = self.get_channel_data(channel)
+
+        # conversions
+        hc = (constants.h * constants.c).to(u.eV * u.angstrom)
+        electron_per_ev = (1 * u.electron) / (3.98 * u.eV)
+
+        if use_table2_gain:
+            # Uses Gain from table Table 12 to calculate instrument response
+            # gain values from Boerner paper
+            gain_table2 = {94: 2.128, 131: 1.523, 171: 1.168, 193: 1.024, 211: 0.946, 304: 0.658, 335: 0.596,
+                           1600: 0.125, 1700: 0.118}
+            self.system_gain = gain_table2[channel]
+
+
+        elif use_genx_values:
+            # calculate the index of values to use from wavelength range
+            channel_index = self.wavelength_range_index(channel)
+
+            # elecperphot in genx starting with the photon energy in eV
+            wavelength = (var['wavelength_range'][channel_index] * u.angstrom)
+            ev_per_wavelength = hc / wavelength
+
+            # converts energy of wavelength from eV to electrons
+            genx_electron_per_ev = var['electron_per_ev'] * (u.count / u.eV)
+            electron_per_wavelength = (ev_per_wavelength * genx_electron_per_ev)
+
+            # gain = elecperphot / elecperdn
+            self.system_gain = electron_per_wavelength / var['electron_per_dn']
+
+
+        else:
+            # calculate entire response using calculated eff_area and ccd gain from table12
+            ccd_gain = {94: 18.3, 131: 17.6, 171: 17.7, 193: 18.3, 211: 18.3, 304: 18.3, 335: 17.6, 1600: 0.125,
+                        1700: 0.118}
+
+            # the photon energy in eV
+            wavelength = (float(channel) * u.angstrom)
+            ev_per_wavelength = hc / wavelength
+
+            # converts energy of wavelength from eV to electrons
+            electron_per_wavelength = (ev_per_wavelength * electron_per_ev)
+
+            # gain of ccd from genx file:
+            ccdgain = ccd_gain[channel] * (u.electron / u.count)
+
+            # Gain(wavelength)  calculation
+            calculated_gain = electron_per_wavelength * (u.count / u.photon)
+
+            self.system_gain = calculated_gain / ccdgain
+
+    def calculate_wavelength_response(self, channel, band_width=2.5,
```

Comment:
Could there also be a function that calculates the wavelength response for each channel in `self.channel_list` and then stores it in some dict `self.wavelength_response_functions` where each key is just the channel word, e.g. `self.wavelength_response_functions['94']` returns the wavelength response function for the 94 angstrom channel.

This would really be just a matter of wrapping the `calculate_wavelength_response` function in a for loop over the channel list.